pandas/tests/extension/decimal/test_decimal.py

```diff
@@ -168,20 +165,32 @@ class TestBooleanReduce(Reduce, base.BaseBooleanReduceTests):
 
 class TestMethods(BaseDecimal, base.BaseMethodsTests):
     @pytest.mark.parametrize("dropna", [True, False])
-    @pytest.mark.xfail(reason="value_counts not implemented yet.")
-    def test_value_counts(self, all_data, dropna):
+    def test_value_counts(self, all_data, dropna, request):
+        if any(x != x for x in all_data):
+            mark = pytest.mark.xfail(
+                reason="tm.assert_series_equal incorrectly raises",
+                raises=AssertionError,
+            )
+            request.node.add_marker(mark)
+
         all_data = all_data[:10]
         if dropna:
             other = np.array(all_data[~all_data.isna()])
         else:
             other = all_data
 
-        result = pd.Series(all_data).value_counts(dropna=dropna).sort_index()
-        expected = pd.Series(other).value_counts(dropna=dropna).sort_index()
+        vcs = pd.Series(all_data).value_counts(dropna=dropna)
+        vcs_ex = pd.Series(other).value_counts(dropna=dropna)
+
+        with decimal.localcontext() as ctx:
+            # avoid raising when comparing Decimal("NAN") < Decimal(2)
```

Comment:
should we be handling this error *inside* the EA e.g. isn't this a reason that DecimalArray should define values_for_argsort?

Reply:
until we have a general ExtensionIndex, values_for_argsort wont help

Reply:
ahh because this is the index of value counts itself.